src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/ShellInProcess.cs

```diff
@@ -8,17 +8,62 @@
 using System.Reflection;
 using System.Threading;
 using System.Threading.Tasks;
+using System.Windows.Controls;
+using System.Windows.Input;
 using Microsoft.CodeAnalysis.UnitTests;
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Shell.Interop;
 using Microsoft.VisualStudio.Text;
 using Microsoft.VisualStudio.Threading;
+using Xunit;
 using IAsyncDisposable = System.IAsyncDisposable;
 
 namespace Microsoft.VisualStudio.Extensibility.Testing
 {
     internal partial class ShellInProcess
     {
+        /// <returns>True if the AllInOneSearch is being used for Navigation</returns>
+        public async Task<bool> ShowNavigateToDialogAsync(CancellationToken cancellationToken)
+        {
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            await TestServices.Shell.ExecuteCommandAsync(VSConstants.VSStd12CmdID.NavigateTo, cancellationToken);
+
+            return await WaitForNavigateToFocusAsync(cancellationToken);
+
+            async Task<bool> WaitForNavigateToFocusAsync(CancellationToken cancellationToken)
+            {
+                bool? isAllInOneSearchActive = null;
+
+                while (true)
+                {
+                    //cancellationToken.ThrowIfCancellationRequested();
+
+                    // Take no direct action regarding activation, but assert the correct item already has focus
+                    TestServices.JoinableTaskFactory.Run(async () =>
+                    {
+                        await TestServices.JoinableTaskFactory.SwitchToMainThreadAsync();
+                        var searchBox = Assert.IsAssignableFrom<Control>(Keyboard.FocusedElement);
+                        if ("PART_SearchBox" == searchBox.Name)
+                        {
+                            isAllInOneSearchActive = false; // Old search name
+                        }
+                        else if ("SearchBoxControl" == searchBox.Name)
+                        {
+                            isAllInOneSearchActive = true; // All-in-one search name
+                        }
+                    });
+
+                    if (isAllInOneSearchActive.HasValue)
+                    {
+                        return isAllInOneSearchActive.Value;
+                    }
+
+                    await Task.Delay(100);
```

Comment:
Ah, can we add a comment indicating this is a workaround?

Reply:
Sure!